### ISSUE TITLE
error when checking for updates http 401

### ISSUE THREAD

**Issue by mohdshara** [nvaccess/nvda#7893]
### Steps to reproduce:
1. from the NVDA menu go to help then check for updates.
2. Error  dialog  Error checking for update.
3. from the error log set to input/output:
DEBUGWARNING - updateCheck.UpdateChecker._bg (12:40:45.154):
Error checking for update
Traceback (most recent call last):
  File "updateCheck.pyo", line 116, in _bg
  File "updateCheck.pyo", line 74, in checkForUpdate
  File "urllib.pyo", line 87, in urlopen
  File "urllib.pyo", line 213, in open
  File "urllib.pyo", line 457, in open_https
  File "urllib.pyo", line 377, in http_error
  File "urllib.pyo", line 642, in http_error_302
  File "urllib.pyo", line 669, in redirect_internal
  File "urllib.pyo", line 213, in open
  File "urllib.pyo", line 457, in open_https
  File "urllib.pyo", line 377, in http_error
  File "urllib.pyo", line 697, in http_error_401
  File "urllib.pyo", line 386, in http_error_default
IOError: ('http error', 401, 'Unauthorized', <httplib.HTTPMessage instance at 0x05817E40>)
IO - tones.beep (12:40:45.157):
Beep at pitch 1760, for 40 ms, left volume 50, right volume 50
IO - speech.speak (12:40:45.213):
Speaking [LangChangeCommand ('en'), u'Error  dialog  Error checking for update.']

### Expected behavior:
nvda should either succeed to check for updates or display more verbose reason for the update failure.

### Actual behavior:
the error quoted above.

### System configuration:
NVDA version: 2017.4

NVDA Installed or portable: installed

Other information:
1. this happens on my work computer which is connected to the internet through a broxy.
2. Downloading nvda using a browser and updating works fine.
3. my domain user is a part of the administrative group on the local computer, so installing and updating software always works.
4. our network detects broxy settings automatically, however, even if I specifically enter the broxy settings through windows I get the same result.

Windows version: windows 10 

Name and version of other software in use when reproducing the issue: nothing of relation to this

### Other questions:

Does the issue still occur after restarting your PC? yes it does

Have you tried any other versions of NVDA? yes, nvda 2017.3 same result as above

**Comment by feerrenrut** [comment]
For someone considering tackling this:

I think this could be addressed in two parts (happy to accept as separate PRs):
- Improve the error detection, and present the error to the user in a friendlier way
- Try to work out why the update check fails in this case and fix. Perhaps start by trying to answer:
  - Is NVDA actually using the proxy?

**Comment by SaschaCowley** [comment]
@mohdshara are you still experiencing this issue?

**Comment by seanbudd** [comment]
Closing. Behaviour has improved a lot here. Given our recent outage - we've been showing messages saying people to check their internet connection while our server is down. @Qchristensen suggested we should rephrase this to say it's possible our server is down, or check if there is a valid internet connection other than the server.
I think a new issue should be opened for any ideas on how to improve the failure message for update check and the add-on store failing.
